### PR TITLE
ORKTableStepViewController improvements

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
These changes to the `ORKTableStepViewController` allow for greater flexibility in the use of an ORKStep that does _not_ inherit from `ORKTableStep` by using a protocol to define the source for the `ORKTableStepViewController`. This allows for an inheritance pattern of using an `ORKFormStep` (for example) with a custom implementation for the table view cells displayed by the form step.

### CODE REVIEW

File: ResearchKit/Common/ORKTableStepViewController.m [hunk 1]
```diff
@@ -46,12 +46,13 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 #import "ORKHelpers_Internal.h"
 #import "ORKSkin.h"
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKTableStepViewController.m]: Add extra new line.

File: ResearchKit/Common/ORKTableStepViewController.m [hunk 2]
```diff
@@ -46,12 +46,13 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
...
 
+ORKDefineStringKey(ORKBasicCellReuseIdentifier);
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKTableStepViewController.m]: Add extra new line.

File: ResearchKit/Common/ORKTableStep.h [hunk 3]
```diff
@@ -36,6 +36,58 @@
...
+    - (NSString *)reuseIdentifierForRowAtIndexPath:(NSIndexPath *)indexPath;
+    
+    /**
+     Optional override for registering UITableViewCell instances. The default registers a `UITableViewCell` 
+     for `ORKBasicCellReuseIdentifier`.
+     
+     @param tableView       The table view to register cells
+     */
+     @optional
+    - (void)registerCellsForTableView:(UITableView *)tableView;
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKTableStep.h]: Fix indentation to match RK coding style.

**Comment by syoung-smallwisdom** [comment]
@umerkhan-apple 
Code style cleaned up.

**Pull request merged**